crates/iroha_core/src/block.rs

```diff
@@ -599,41 +572,56 @@ mod valid {
             Ok(())
         }
 
-        fn validate_transactions(
+        fn categorize(
             block: &mut SignedBlock,
             expected_chain_id: &ChainId,
             genesis_account: &AccountId,
             state_block: &mut StateBlock<'_>,
         ) -> Result<(), TransactionValidationError> {
```

Comment:
what can I do about it?

Reply:
Considering the current usage, I think it could be a `SignedBlock` method that takes `self` instead of a reference and return `Result<ValidBlock, _>`

Reply:
ah, the problem is that then I have to define a trait in the `core` crate because I cannot add an impl for a trait that's defined in the `data_model` crate

Reply:
Ok, then what if make it a `ValidBlock` method that takes `SignedBlock` and return `Result<Self, _>`? The idea is changing type rather than mutating the same struct through validation

Reply:
This is what happens generally, but  this is a private method